packages/flutter_tools/lib/src/template.dart

```diff
@@ -322,6 +322,16 @@ class Template {
 
       if (sourceFile.path.endsWith(templateExtension)) {
         final String templateContents = sourceFile.readAsStringSync();
+        final String? androidIdentifier = context['androidIdentifier'] as String?;
+        if (finalDestinationFile.path.endsWith('.kt') && androidIdentifier != null) {
+          final List<String> segments = androidIdentifier.split('.');
+          final List<String> reserved = <String>['when', 'in'];
+          final List<String> correctedSegments = segments.map(
+              (String segment) => reserved.contains(segment) ? '`$segment`' : segment
```

Comment:
I think this is the wrong spot to perform the check. I would move this to create_base where the identifier is parsed.

For the kotlin reserved words, this should be in a top level constant with a link to documentation so we can keep it up to date.

Reply:
The `androidIdentifier` is used for multiple purposes, e.g. creating the folder, in `android\app\src\main\kotlin` and for the package name inside the `MainActivity.kt`.

Putting the logic globally (inside the `create_base`), would make a folder named with \`in\` which is not what is needed AFAIHK.

The reserved keywords were put in a static top-level field.